### PR TITLE
RHIROS-366 | ADD - Updated .gitignore

### PR DESCRIPTION
**Pull request by saltgen**
Changes:
1. File updated

Reasons:
1. As discussed on https://github.com/RedHatInsights/ros-backend/pull/122#pullrequestreview-799898460

Additional:
As per suggestions from [gitignore.io](https://gitignore.io) with Pycharm and VisualStudioCode filters

### CODE REVIEW

**Comment by kgaikwad** [comment]
@PreetiW & @r14chandra - As you are using the same editor, please request to review this .gitignore file change.

**Review by r14chandra — APPROVED** [review]
LGTM @saltgen. Tested the PR locally. 💯

File: .gitignore [hunk 1]
```diff
@@ -133,3 +133,30 @@ dmypy.json
...
+!.vscode/launch.json
+!.vscode/extensions.json
+*.code-workspace
+
+# Local History for Visual Studio Code
+.history/
+
+### VisualStudioCode Patch ###
+# Ignore all local history of files
+.history
```
Comment (kgaikwad) [inline .gitignore:158]: @saltgen - does that mean both directory and file are present?

Comment (saltgen) [reply .gitignore]: Still not sure if VS Code creates both `.history` folder and file. 

Regardless, tested locally by creating `.history`(non-empty) folder and another `.history` file. 

`.history` entry on .gitignore takes care of both, changes pushed :+1:

**Comment by apuntamb** [comment]
Hi, @saltgen ! The PR looks ready to go. 
I just have one question before we merge this. If you can view my local [.gitignore](https://github.com/RedHatInsights/ros-backend/compare/main...apuntamb:dummy-gitignore-vs-code?expand=1), I can see pretty large git diff here.
However, I'm not much aware about the gitignore changes, I have simply used a gitignore generator extension provided by vscode. Let me know what you think about it. If it doesn't make much sense, we can definitely ignore my shared diffs. Feel free to share your thoughts.

**Comment by saltgen** [comment]
> Hi, @saltgen ! The PR looks ready to go. I just have one question before we merge this. If you can view my local [.gitignore](https://github.com/RedHatInsights/ros-backend/compare/main...apuntamb:dummy-gitignore-vs-code?expand=1), I can see pretty large git diff here. However, I'm not much aware about the gitignore changes, I have simply used a gitignore generator extension provided by vscode. Let me know what you think about it. If it doesn't make much sense, we can definitely ignore my shared diffs. Feel free to share your thoughts.

Just checked, yeah that is a lot of diffs :sweat_smile: 
I think we include them apart from `eclipse stuff, qcreator, and Catkin custom files, .history/`
Not sure about `ROS` though :thinking: 
Thoughts? @kgaikwad 
Diff PR [here](https://github.com/RedHatInsights/ros-backend/pull/133)

**Comment by apuntamb** [comment]
@saltgen yeah we can ignore the ROS diff.
Also, closing the [dummy PR](https://github.com/RedHatInsights/ros-backend/pull/133) to avoid any confusions in the PR list as that was meant to be just a commit in my forked repo.

**Comment by saltgen** [comment]
> @saltgen yeah we can ignore the ROS diff. Also, closing the [dummy PR](https://github.com/RedHatInsights/ros-backend/pull/133) to avoid any confusions in the PR list as that was meant to be just a commit in my forked repo.

Changes pushed :+1:  @apuntamb @kgaikwad

File: .gitignore [hunk 2]
```diff
@@ -128,8 +131,50 @@ venv.bak/
...
+.fuse_hidden*
+
+# KDE directory preferences
+.directory
+
+# Linux trash folder which might appear on any partition or disk
+.Trash-*
+
+# .nfs files are created when an open file is removed but is still being accessed
+.nfs*
```
Comment (kgaikwad) [inline .gitignore:180]: @saltgen - could you please include newline at the end?

Comment (saltgen) [reply .gitignore]: Added, change pushed

**Pull request merged**